TheJumpCloud/support
# Sa 2022 add code artifact to pwsh pipeline

## Issues
* [SA-2022](https://jumpcloud.atlassian.net/browse/SA-2022) - Add CodeArtifact support to JumpCloud PowerShell Module CI Workflow

## What does this solve?
If ever the JumpCloud PowerShell SDKs Modules need to release in tandem with the JumpCloud PowerShell Module release, this PR allows us to do just that. This PR allows us to test the prerelease SDK modules in our CI pipeline so we can release both the SDK modules and the JumpCloud module together.

## Is there anything particularly tricky?
Yes, instead of an environment variable the $RequiredModulesRepo variable is now set as a parameter in circleCI. Almost every file that referenced the $RequiredModulesRepo env variable this now has a parameter block to set the $RequiredModulesRepo.

Switching between pulling the latest production SDK modules and whatever is latest on CodeArtifact is now done through the RequiredModuelsRepo Parameter in workflow.yml. 

In order to use the CodeArtifact Nuget v3 repository we are using a beta version of PowerShellGet. There are a few interesting things with the new PowerShellGet: Modules are now installed with `Install-PSResource` 

## How should this be tested?

See [Build 311](https://app.circleci.com/pipelines/github/TheJumpCloud/support/311/workflows/28781f92-54b6-49ac-86f9-348ab34d39d5) For an example of the pipeline running with RequiredModuleRepo set to PSModule

See [Build 309](https://app.circleci.com/pipelines/github/TheJumpCloud/support/309/workflows/612dbdcb-006a-4248-abaa-ea06d7d9e364) For an example of the pipeline running with RequiredModuleRepo set to CodeArtifact

## Screenshots

## Notes:
The only reason Get-jcEvent & Get-JCEvent count are included in this PR is spacing changes, I can revert those along side ModuleChangelog ModuleBanner & JumpCloud.ps1 (Module Version Change) so we can release to Master w/o doing a release
